src/views/News/index.js

```diff
@@ -19,530 +23,764 @@ import {
   DialogActions,
   DialogContent,
   DialogTitle,
+  DialogContentText,
   MenuItem,
 } from '@material-ui/core';
 import CloseIcon from '@material-ui/icons/Close';
 import { ReactComponent as NoConnectionImage } from 'NoConnection.svg';
-// testing for future feature to upload image
-// import IDUploader from '../IDUploader';
-// import Dropzone from 'react-dropzone';
-
-const styles = {
-  searchBar: {
-    margin: '0 auto',
-  },
-  newNewsForm: {
-    backgroundColor: '#fff',
-  },
-  fab: {
-    margin: 0,
-    top: 'auto',
-    right: 40,
-    bottom: 40,
-    left: 'auto',
-    position: 'fixed',
-    zIndex: 1,
-  },
-};
+import useNetworkStatus from 'hooks/useNetworkStatus';
+
+const BREAKPOINT_WIDTH = 540;
+const CROP_DIM = 200; // pixels
+
+const StudentNews = (props) => {
+  const [search, setSearch] = useState('');
+  const [openPostActivity, setOpenPostActivity] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [categories, setCategories] = useState([]);
+  const [news, setNews] = useState([]);
+  const [personalUnapprovedNews, setPersonalUnapprovedNews] = useState([]);
+  //const [filteredNews, setFilteredNews] = useState([]);
+  const isOnline = useNetworkStatus();
+  const [newPostCategory, setNewPostCategory] = useState('');
+  const [newPostSubject, setNewPostSubject] = useState('');
+  const [newPostBody, setNewPostBody] = useState('');
+  const [showCropper, setShowCropper] = useState(null); //null if no picture chosen, else contains picture
+  const [photoDialogErrorTimeout, setPhotoDialogErrorTimeout] = useState(null);
+  const [photoDialogError, setPhotoDialogError] = useState(null);
+  const [cropBoxDim, setCropBoxDim] = useState(null);
+  const [aspectRatio, setAspectRatio] = useState(null);
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+  const [snackbarMessage, setSnackbarMessage] = useState('Something went wrong');
```

Comment:
The snackbar logic in this can be drastically simplified and improved, using our custom `GordonSnackbar` component and clever use of `setState`. Checkout `src/views/InvolvementProfile/components/Membership/index.js` lines 19, 50-52, and 149-153 for an example.